tty.c

```diff
@@ -109,101 +116,232 @@ void tty_setup_socketpair(int *pParentSock, int *pChildSock) {
 	*pChildSock = socks[1];
 }
 
+static ssize_t io_copy(int out_fd, int in_fd, struct buffer *buf)
+{
+	ssize_t copied = 0;
+	ssize_t rd = -1;
+
+	for (;;) {
+
+		/* Write any leftover data from a previous read. This handles the case
```

Comment:
I have a set of changes to use splice instead of an internal buffer.  This might make this case easier since it will use the pipe's internal buffers.